src/parser/expression.js

```diff
@@ -613,6 +613,10 @@ pp.parseParenAndDistinguishExpression = function (startPos, startLoc, canBeArrow
   return val;
 };
 
+pp.shouldParseArrow = function () {
+	return !this.canInsertSemicolon();
```

Comment:
Oops, sorry! (I'm one of _those_ people lol). Lint didn't pick it up. I've pushed a change.